src/power_grid_model/data_types.py

```diff
@@ -92,39 +92,50 @@
     - The first element and last element will therefore be 0 and the size of the data, respectively.
 """
 
-SparseBatchArray = dict[SparseDataComponentType, IndexPointer | SingleArray]
-"""
-A sparse batch array is a dictionary containing the keys `indptr` and `data`.
 
-- data: a :class:`SingleArray`. The exact dtype depends on the type of component.
-- indptr: an :class:`IndexPointer` representing the start and end indices for each batch scenario.
+class SparseBatchArray(TypedDict):
```

Comment:
Special attention here. Maybe doing this is not okay

Reply:
Should a Todo be added here for that matter? I am not sure what to make of this comment

Reply:
Dont think there are other places for it. Lets move forward with this for now then.